### DISCUSSION TITLE
How do I use a SET statement in a cluster

### DISCUSSION THREAD

**Discussion by Darcy-Sun** [ClickHouse/ClickHouse#40924] [Q&A]
How do I use a SET statement in a cluster
when I use "set opentelemetry_start_trace_probability=1 " in cluster 

will appear:
SQL error [113] [07000]: Code: 113. DB::Exception: There is no session or session context has expired. (THERE_IS_NO_SESSION) (version 21.12.3.1)
, server ClickHouseNode(addr=http:1.116.223.245:9123, db=default)@-924811823

I need use opentelemetry_start_trace_probability
 But when I use  'select 1 settings opentelemetry_start_trace_probability=1'  I can't find message from zipkin,  when I use 'show tables ( or drop , create )settings opentelemetry_start_trace_probability=1' I can find message.

**Comment by UnamedRus** [comment]
You can append this setting to end of query in SETTINGS clause.

Or define in user profile.

**Comment by qoega** [comment]
Open telemetry is enabled before SETTINGS are parsed in query, so you have to enable it on session level. In client arguments '--opentelemetry_start_trace_probability=1' or in HTTP query '&opentelemetry_start_trace_probability=1'
